composer/datasets/lm_datasets.py

```diff
@@ -85,13 +85,38 @@ def initialize_object(self, batch_size: int, dataloader_hparams: DataloaderHpara
                 if using pip or `conda install -c conda-forge datasets transformers` if using Anaconda.""")) from e
 
         self.validate()
-        self.tokenizer = transformers.AutoTokenizer.from_pretrained(self.tokenizer_name)  #type: ignore (thirdparty)
-        self.config = transformers.AutoConfig.from_pretrained(self.tokenizer_name)  #type: ignore (thirdparty)
-        lm_datasets = [datasets.load_from_disk(i) for i in self.datadir]  #type: ignore (thirdparty)
+        assert self.tokenizer_name is not None
+        if self.use_synthetic:
+            column_names = ["text"]
+
+            # we just use the max sequence length in tokens to upper bound the sequence length in characters
+            lm_datasets = SyntheticHFDataset(num_samples=self.synthetic_num_unique_samples,
+                                             chars_per_sample=self.max_seq_length,
+                                             column_names=column_names).generate_dataset()
+
+            tokenizer = generate_synthetic_tokenizer(tokenizer_family=self.tokenizer_name, dataset=lm_datasets)
+
+            columns_to_remove = ["idx"] + column_names
+            lm_datasets = lm_datasets.map(lambda inp: tokenizer(
+                text=inp[column_names[0]], padding="max_length", max_length=self.max_seq_length, truncation=True),
+                                          batched=True,
+                                          num_proc=1,
```

Comment:
should this be a variable instead of 1?

Reply:
imo, the synthetic dataset is always trivially small. I can make it a variable, but a synthetic dataset, by definition, shouldn't require too many processes to preprocess.

I am open to changing it, but curious for your thoughts given the context!

Reply:
Could you use `dataloader_hparams.num_workers`? I think that should be a reasonable value for here.

Reply:
@ravi-mosaicml done!